### PR TITLE
[Storybook] Add ability to take snapshots during Storybook `play` interactions

### PR DESCRIPTION
**Pull request by omacranger**
## Description
This PR adds the ability to take snapshots _during_ Storybook play interactions, instead of just at the end. This also reintroduces Storybook 6 support by tweaking dependency requirements.

For example, you could now open a dropdown, focus an input field, type text to generate form errors all while taking snapshots at each step of the way to catch all visual edge cases without creating individual stories for each.

This requires a new `preVisit` hook which has to be added to the Storybook test runner configuration. If attempting to use the `takeVisualSnapshot` without it, it will throw an error and mention this new addition.

## Types of Changes

- New feature (non-breaking change which adds functionality)
- Configuration change

### CODE REVIEW

File: visual-js/visual-storybook/src/api.ts [hunk 1]
```diff
@@ -143,3 +145,32 @@ export const postVisit = async (page: Page, context: TestContext) => {
  * `postVisit` exported from this package instead.
  */
 export const postRender = postVisit;
+
+let hasExposed = false;
```
Comment (ebekebe) [inline visual-js/visual-storybook/src/api.ts:149]: Could you add a comment about the intent and lifecycle of this?

Comment (omacranger) [reply visual-js/visual-storybook/src/api.ts:156]: Added

**Review by ebekebe — APPROVED** [review]
